src/EditorFeatures/Core.Wpf/Suggestions/FixAll/FixAllCodeFixGetFixesService.cs

```diff
@@ -0,0 +1,110 @@
+            {
+                return fixAllContext.Solution;
+            }
+
+            fixAllContext.CancellationToken.ThrowIfCancellationRequested();
+            return await codeAction.GetChangedSolutionInternalAsync(cancellationToken: fixAllContext.CancellationToken).ConfigureAwait(false);
+        }
+
+        public async Task<ImmutableArray<CodeActionOperation>> GetFixAllOperationsAsync(
+            FixAllContext fixAllContext, bool showPreviewChangesDialog)
+        {
+            var codeAction = await GetFixAllCodeActionAsync(fixAllContext).ConfigureAwait(false);
+            if (codeAction == null)
+            {
+                return ImmutableArray<CodeActionOperation>.Empty;
+            }
+
+            return await FixAllGetFixesServiceHelper.GetFixAllOperationsAsync(
+                codeAction, fixAllContext.State.Project, fixAllContext.State.CorrelationId,
+                FunctionId.CodeFixes_FixAllOccurrencesPreviewChanges,
+                showPreviewChangesDialog, fixAllContext.CancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<CodeAction> GetFixAllCodeActionAsync(FixAllContext fixAllContext)
+        {
+            using (Logger.LogBlock(
+                FunctionId.CodeFixes_FixAllOccurrencesComputation,
+                KeyValueLogMessage.Create(LogType.UserAction, m =>
+                {
+                    m[FixAllLogger.CorrelationId] = fixAllContext.State.CorrelationId;
+                    m[FixAllLogger.FixAllScope] = fixAllContext.State.Scope.ToString();
+                }),
+                fixAllContext.CancellationToken))
+            {
+                CodeAction action = null;
+                try
+                {
+                    action = await fixAllContext.FixAllProvider.GetFixAsync(fixAllContext).ConfigureAwait(false);
+                }
+                catch (OperationCanceledException)
+                {
+                    FixAllLogger.LogComputationResult(fixAllContext.State.CorrelationId, completed: false);
+                }
+                finally
+                {
+                    if (action != null)
+                    {
+                        FixAllLogger.LogComputationResult(fixAllContext.State.CorrelationId, completed: true);
+                    }
+                    else
+                    {
+                        FixAllLogger.LogComputationResult(fixAllContext.State.CorrelationId, completed: false, timedOut: true);
+                    }
+                }
+
+                return action;
+            }
+        }
+
+        internal static Solution PreviewChanges(
+            Solution currentSolution,
+            Solution newSolution,
+            string fixAllPreviewChangesTitle,
+            string fixAllTopLevelHeader,
+            string languageOpt,
+            Workspace workspace,
+            int? correlationId = null,
+            CancellationToken cancellationToken = default)
```

Comment:
please don't make internal methods have optional cancellation tokens.  That just means it's easy for us to forget to pass them along.

Reply:
Existing code, I just refactored it move out common code to a shared type.